pghoard/archive_sync.py

```diff
@@ -199,7 +198,7 @@ def check_wal_archive_integrity(self, new_backup_on_failure):
             # Go back one timeline and flag the current timeline as invalid, this will prevent segment
             # number from being decreased on the next iteration.
             valid_timeline = False
-            current_tli -= 1
+            current_lsn = current_lsn.at_timeline(current_lsn.timeline_id - 1)
```

Comment:
BTW: this looks like a code change (bug fix?), not only a 1:1 transfer. 

(I think we have an internal issues with failover and RR not coming up, that's why I'm curious...)

Reply:
If that's a code change, it's not intentional... I wanted to have just one place where we store a timeline: associated to the LSN in this case.

But I agree something is fishy with that code, I wanted to add a comment before noticing there already was one just above:

```
# TODO: Need to check .history files as well
```
I think we should not rely on the previous timeline being one with an id = current_timeline_id - 1: we should use the timeline history to determine what timeline the current one has been created from.

Reply:
... after reading the code behind at_timeline: ok, seems to be the same :-)